src/numbuf/cpp/src/numbuf/sequence.h

```diff
@@ -3,10 +3,19 @@
 
 #include "tensor.h"
 #include <arrow/api.h>
-#include <arrow/types/union.h>
+#include <arrow/array.h>
```

Comment:
this is in `arrow/api.h` -- I recommend only using the `api.h` headers -- if you hit a missing include please open a JIRA/make a PR